draft-ietf-tls-tls13.md

```diff
@@ -1566,7 +1566,9 @@ cipher_suites
   order of client preference. If the list contains cipher suites
   the server does not recognize, support, or wish to use, the server
   MUST ignore those cipher suites, and process the remaining ones as
-  usual. Values are defined in {{cipher-suites}}.
+  usual. Values are defined in {{cipher-suites}}. If the client is
+  attempting a PSK key establishment, it MUST only advertise the single
+  cipher suite associated with the PSK.
```

Comment:
The server is responsible for only picking PSK if it also selects a compatible cipher suite. So the situation you describe is a bug in the server. The critical concern is that the handshake has to complete successfully if at all possible. If you compute the wrong key, get a MAC error and retry, then you have problems.

Consider the case where the client offers A and B, and the server selects A. If the client's attempt to resume only includes A, then if the server has disabled A, then you will have a negotiation failure even if you could have connected. By contrast, if the client offers both A and B but the PSK can only be used with A, then if the server has disabled B then you just get no resumption but B.